crates/debugger/src/tui/draw.rs

```diff
@@ -350,7 +352,20 @@ impl DebuggerContext<'_> {
                 let pc_ic_map = if is_create { create_map } else { rt_map };
                 let ic = pc_ic_map.get(pc)?;
                 let source_element = source_map.swap_remove(ic);
-                (*file_id == source_element.index?).then_some((source_element, source_code))
+                // if the source element has an index, find the sourcemap for that index
+                source_element.index.and_then(|index| {
+                    // if index matches current file_id, return current source code
+                    (index == *file_id).then_some((source_element.clone(), source_code)).or_else(
+                        || {
+                            // otherwise find the source code for the element's index
+                            self.debugger
+                                .contracts_sources
+                                .sources_by_id
+                                .get(&index)
+                                .map(|(source_code, _)| (source_element.clone(), source_code))
+                        },
+                    )
+                })
```

Comment:
this suggestion is poorly formatted, but in essence I think you should be able to remove one layer of nesting by pulling the `or_else` out

```suggestion
                 // if index matches current file_id, return current source code
                source_element.index.and_then(|index| (index == *file_id).then_some((source_element.clone(), source_code))).or_else(
                        || {
                            // otherwise find the source code for the element's index
                            self.debugger
                                .contracts_sources
                                .sources_by_id
                                .get(&index)
                                .map(|(source_code, _)| (source_element.clone(), source_code))
                        },
                    );
```

Reply:
Cool – implemented; need to unwrap the `source_element.index` again since I was using the closure value originally.